From 26c539146ceea6a42b7b33b798289d25f4fbfc9b Mon Sep 17 00:00:00 2001
From: William Wu <william.wu@rock-chips.com>
Date: Sat, 24 Feb 2018 16:52:02 +0800
Subject: [PATCH] ARM: dts: rockchip: use regulator for otg vbus on rk3229
 boards

This patch uses a fixed-regulator instead of GPIO pin for
otg vbus power on rk3229 boards.

Change-Id: I9b4f37c78a8d112008c3db552150a77bbaeb23fc
Signed-off-by: William Wu <william.wu@rock-chips.com>
---
 arch/arm/boot/dts/rk3229-at-som.dtsi     | 18 +++++++++++++++++-
 arch/arm/boot/dts/rk3229-echo-v10.dts    | 19 +++++++++++++++++++
 arch/arm/boot/dts/rk3229-evb-android.dts | 19 +++++++++++++++++++
 arch/arm/boot/dts/rk3229-evb.dts         | 19 +++++++++++++++++++
 arch/arm/boot/dts/rk3229-gva-sdk.dts     | 19 +++++++++++++++++++
 arch/arm/boot/dts/rk322x-android.dtsi    |  1 -
 6 files changed, 93 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/rk3229-at-som.dtsi b/arch/arm/boot/dts/rk3229-at-som.dtsi
index 27ffe812b39a..4ff30be58aae 100644
--- a/arch/arm/boot/dts/rk3229-at-som.dtsi
+++ b/arch/arm/boot/dts/rk3229-at-som.dtsi
@@ -79,6 +79,16 @@
 		};
 	};
 
+	vcc_otg_vbus: otg-vbus-regulator {
+		compatible = "regulator-fixed";
+		gpio = <&gpio3 RK_PC6 GPIO_ACTIVE_HIGH>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&otg_vbus_drv>;
+		regulator-name = "vcc_otg_vbus";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		enable-active-high;
+	};
 };
 
 &hdmi {
@@ -197,6 +207,12 @@
 			rockchip,pins = <3 31 RK_FUNC_GPIO &pcfg_pull_none>;
 		};
 	};
+
+	usb {
+		otg_vbus_drv: otg-vbus-drv {
+			rockchip,pins =	<3 RK_PC6 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+	};
 };
 
 &rockchip_suspend {
@@ -242,10 +258,10 @@
 };
 
 &u2phy0 {
-	otg-vbus-gpios = <&gpio3 RK_PC6 GPIO_ACTIVE_HIGH>;
 	status = "okay";
 
 	u2phy0_otg: otg-port {
+		vbus-supply = <&vcc_otg_vbus>;
 		status = "okay";
 	};
 
diff --git a/arch/arm/boot/dts/rk3229-echo-v10.dts b/arch/arm/boot/dts/rk3229-echo-v10.dts
index c539e5671696..de70e93b1620 100644
--- a/arch/arm/boot/dts/rk3229-echo-v10.dts
+++ b/arch/arm/boot/dts/rk3229-echo-v10.dts
@@ -93,6 +93,17 @@
 		regulator-boot-on;
 	};
 
+	vcc_otg_vbus: otg-vbus-regulator {
+		compatible = "regulator-fixed";
+		gpio = <&gpio3 RK_PC6 GPIO_ACTIVE_HIGH>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&otg_vbus_drv>;
+		regulator-name = "vcc_otg_vbus";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		enable-active-high;
+	};
+
 	wireless-bluetooth {
 		compatible = "bluetooth-platdata";
 		clocks = <&rk805 1>;
@@ -305,6 +316,10 @@
 		host_vbus_drv: host-vbus-drv {
 			rockchip,pins = <3 20 RK_FUNC_GPIO &pcfg_pull_none>;
 		};
+
+		otg_vbus_drv: otg-vbus-drv {
+			rockchip,pins = <3 RK_PC6 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
 	};
 
 	keys {
@@ -330,6 +345,10 @@
 	phy-supply = <&vcc_host>;
 };
 
+&u2phy0_otg {
+	vbus-supply = <&vcc_otg_vbus>;
+};
+
 &usb_host0_ehci {
 	status = "okay";
 };
diff --git a/arch/arm/boot/dts/rk3229-evb-android.dts b/arch/arm/boot/dts/rk3229-evb-android.dts
index df57a1b6349d..fa177a7c1f07 100644
--- a/arch/arm/boot/dts/rk3229-evb-android.dts
+++ b/arch/arm/boot/dts/rk3229-evb-android.dts
@@ -127,6 +127,17 @@
 		regulator-boot-on;
 	};
 
+	vcc_otg_vbus: otg-vbus-regulator {
+		compatible = "regulator-fixed";
+		gpio = <&gpio3 RK_PC6 GPIO_ACTIVE_HIGH>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&otg_vbus_drv>;
+		regulator-name = "vcc_otg_vbus";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		enable-active-high;
+	};
+
 	vcc_phy: vcc-phy-regulator {
 		compatible = "regulator-fixed";
 		enable-active-high;
@@ -290,6 +301,10 @@
 		host_vbus_drv: host-vbus-drv {
 			rockchip,pins = <3 20 RK_FUNC_GPIO &pcfg_pull_none>;
 		};
+
+		otg_vbus_drv: otg-vbus-drv {
+			rockchip,pins = <3 RK_PC6 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
 	};
 };
 
@@ -445,6 +460,10 @@
 	phy-supply = <&vcc_host>;
 };
 
+&u2phy0_otg {
+	vbus-supply = <&vcc_otg_vbus>;
+};
+
 &usb_host0_ehci {
 	status = "okay";
 };
diff --git a/arch/arm/boot/dts/rk3229-evb.dts b/arch/arm/boot/dts/rk3229-evb.dts
index ffe32522e3ec..f60d2a974654 100644
--- a/arch/arm/boot/dts/rk3229-evb.dts
+++ b/arch/arm/boot/dts/rk3229-evb.dts
@@ -178,6 +178,17 @@
 		regulator-boot-on;
 	};
 
+	vcc_otg_vbus: otg-vbus-regulator {
+		compatible = "regulator-fixed";
+		gpio = <&gpio3 RK_PC6 GPIO_ACTIVE_HIGH>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&otg_vbus_drv>;
+		regulator-name = "vcc_otg_vbus";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		enable-active-high;
+	};
+
 	vcc_phy: vcc-phy-regulator {
 		compatible = "regulator-fixed";
 		enable-active-high;
@@ -312,6 +323,10 @@
 		host_vbus_drv: host-vbus-drv {
 			rockchip,pins = <3 20 RK_FUNC_GPIO &pcfg_pull_none>;
 		};
+
+		otg_vbus_drv: otg-vbus-drv {
+			rockchip,pins = <3 RK_PC6 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
 	};
 };
 
@@ -347,6 +362,10 @@
 	phy-supply = <&vcc_host>;
 };
 
+&u2phy0_otg {
+	vbus-supply = <&vcc_otg_vbus>;
+};
+
 &usb_host0_ehci {
 	status = "okay";
 };
diff --git a/arch/arm/boot/dts/rk3229-gva-sdk.dts b/arch/arm/boot/dts/rk3229-gva-sdk.dts
index 3f89a8f5a5e7..b10c3cdcf0cd 100644
--- a/arch/arm/boot/dts/rk3229-gva-sdk.dts
+++ b/arch/arm/boot/dts/rk3229-gva-sdk.dts
@@ -139,6 +139,16 @@
 		};
 	};
 
+	vcc_otg_vbus: otg-vbus-regulator {
+		compatible = "regulator-fixed";
+		gpio = <&gpio3 RK_PC6 GPIO_ACTIVE_HIGH>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&otg_vbus_drv>;
+		regulator-name = "vcc_otg_vbus";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		enable-active-high;
+	};
 };
 
 &gpu {
@@ -373,6 +383,11 @@
 		};
 	};
 
+	usb {
+		otg_vbus_drv: otg-vbus-drv {
+			rockchip,pins = <3 RK_PC6 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+	};
 };
 
 &sdio {
@@ -392,3 +407,7 @@
 		)
 	>;
 };
+
+&u2phy0_otg {
+	vbus-supply = <&vcc_otg_vbus>;
+};
diff --git a/arch/arm/boot/dts/rk322x-android.dtsi b/arch/arm/boot/dts/rk322x-android.dtsi
index 29b0f444663f..a0df1ab3a44e 100644
--- a/arch/arm/boot/dts/rk322x-android.dtsi
+++ b/arch/arm/boot/dts/rk322x-android.dtsi
@@ -179,7 +179,6 @@
 };
 
 &u2phy0 {
-	otg-vbus-gpios = <&gpio3 RK_PC6 GPIO_ACTIVE_HIGH>;
 	status = "okay";
 
 	u2phy0_otg: otg-port {
-- 
2.35.3

